gms/gossiper.hh

```diff
@@ -244,8 +245,20 @@ private:
 
     bool _in_shadow_round = false;
 
-    std::unordered_map<inet_address, clk::time_point> _shadow_unreachable_endpoints;
-    utils::chunked_vector<inet_address> _shadow_live_endpoints;
+    uint64_t _shadow_live_endpoints_version = 0;
+    uint64_t _shadow_unreachable_endpoints_version = 0;
+
+    // Must be called on shard 0.
+    future<semaphore_units<>> lock_endpoint_update_semaphore();
+
+    // Must be called on shard 0.
+    // Safely update _live_endpoints and/or _unreachable_endpoints
```

Comment:
I want to mention that even with mutate_live_and_unreachable_endpoints, we still could have inconsistency of _live_endpoints on difference shards. Yes, after this patch, we modify _live_endpoints under the lock. But we do not read _live_endpoints under the lock. So different shards could have different view of  _live_endpoints.

Reply:
True, whoever needs momentary cross-shard consistency can acquire the lock (like `get_live_members_synchronized`)

Reply:
okay